### PR TITLE
Transparent / Gradual Increase to RAM Supply

### PR DESCRIPTION
**Pull request by bytemaster**
Add setramrate method for gradual increase

     - this change allows the elected producers to set the ramrate as bytes
     per block
     - this will also resync the bancor parameters to current eosio.ram
     balance plus reduce the volatility of the market maker
     - added revision field so that future updates to system contract can
     reference it if necessary.
     - updated ABI

### CODE REVIEW

**Comment by sergioyuhjtman** [comment]
By changing the weights in the connector but not the formulas in exchange_state.cpp the exchange between EOS and RAM will remain unaffected (assuming equal weights).

**Comment by hyunwoongJi** [comment]
As @sergioyuhjtman said, changing the weight in the connector does not effect anything to current model

**Comment by bezalel** [comment]
Price-move seems same for CW 50% and CW 0.05%.
But if we use CW 50%, we can optimize token conversion operations.

For CW 0.5%, std:pow(xxx, 0.005) and std:pow(xxx, 200) should be calculated,
but for CW 50%, std:pow(xxx, 0.5) and std:pow(xxx, 2) can be optimized to "sqrt()" and "floating point 
multiplication" which is way more efficient.

So I think CW 50% is better option regarding performance optimization.

**Comment by jonnyhsy** [comment]
@sergioyuhjtman  wan't it done in void system_contract::update_ram_supply(), and void system_contract::setramrate( uint16_t bytes_per_block ) ? _self.base.weight and _self.quote.weight are updated to 500

File: eosio.system/abi/eosio.system.abi [hunk 1]
```diff
@@ -240,7 +240,14 @@
          {"name":"max_inline_action_size",              "type":"uint32"},
          {"name":"max_inline_action_depth",             "type":"uint16"},
          {"name":"max_authority_depth",                 "type":"uint16"}
-
+      ]
+    },{
+      "name": "eosio_global_state2",
+      "fields": [
+         {"name":"new_ram_per_block",   "type":"uint16"},
+         {"name":"last_ram_increase",   "type":"block_timestamp"},
```
Comment (hdennen) [inline eosio.system/abi/eosio.system.abi]: timestamp field names could be more descriptive as well as the `state2` naming. Introduces a naming pattern that could be misappropriated later.

**Comment by zorba80** [comment]
Changes in this PR were moved to PR #9.

**Pull request merged**